spec/datadog/tracing/context_provider_spec.rb

```diff
@@ -106,59 +106,74 @@ def thread_contexts
   end
 
   describe '#local' do
-    subject(:local) { thread_local_context.local }
+    subject(:local) { fiber_local_context.local }
 
-    context 'with a second ThreadLocalContext' do
-      let(:thread_local_context2) { described_class.new }
+    context 'with a second FiberLocalContext' do
+      let(:fiber_local_context2) { described_class.new }
 
-      it 'does not interfere with other ThreadLocalContext' do
-        local_context = thread_local_context.local
-        local_context2 = thread_local_context2.local
+      it 'does not interfere with other FiberLocalContext' do
+        local_context = fiber_local_context.local
+        local_context2 = fiber_local_context2.local
 
         expect(local_context).to_not eq(local_context2)
-        expect(thread_local_context.local).to eq(local_context)
-        expect(thread_local_context2.local).to eq(local_context2)
+        expect(fiber_local_context.local).to eq(local_context)
+        expect(fiber_local_context2.local).to eq(local_context2)
+      end
+    end
+
+    context 'in another fiber' do
+      it 'create one fiber-local variable per fiber' do
+        context = fiber_local_context.local
+
+        Fiber.new do
+          expect { @fiber_context = fiber_local_context.local }
+            .to change { thread_contexts.size }.from(0).to(1)
+
+          expect(@fiber_context).to be_a Datadog::Tracing::Context
+        end.resume
+
+        expect(@fiber_context).to_not eq(context)
```

Comment:
Minor: The instance variable use seems a bit weird to me, and I don't quite like that if somehow the fiber doesn't assign to `@fiber_context`, the `expect(@fiber_context).to_not eq(context)` will still pass (because it'll be `nil`).

So here's a tiny suggestion that I think covers both 😄:  

```suggestion
        main_fiber_context = fiber_local_context.local
        other_fiber_context = nil

        Fiber.new do
          expect { other_fiber_context = fiber_local_context.local }
            .to change { thread_contexts.size }.from(0).to(1)
        end.resume

        expect(other_fiber_context).to be_a Datadog::Tracing::Context
        expect(other_fiber_context).to_not eq(main_fiber_context)
```

Reply:
Nice improvement! I applied it to both places. 🙇